### PR TITLE
[New recipe] Ldmicroqt

### PR DESCRIPTION
**Pull request by mazbrili**
LDmicro is a ladder logic editor, simulator and compiler for 8-bit
microcontrollers.

### CODE REVIEW

File: sci-electronics/ldmicroqt/patches/ldmicroqt-2.3.2.patchset [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+diff --git a/ldmicro/lib/linuxUI/linuxUI.h b/ldmicro/lib/linuxUI/linuxUI.h
+index 356dca6..11e9062 100644
+--- a/ldmicro/lib/linuxUI/linuxUI.h
++++ b/ldmicro/lib/linuxUI/linuxUI.h
+@@ -30,7 +30,7 @@
+ // #include <QSize>
+ // #include "freezeLD.h"
+ // #include "linuxLD.h"
+-#include <linux/limits.h>
++//#include <linux/limits.h>
```
Comment (korli) [inline sci-electronics/ldmicroqt/patches/ldmicroqt-2.3.2.patchset:56]:
```suggestion
#ifndef __HAIKU__
#include <linux/limits.h>
#endif
```

File: sci-electronics/ldmicroqt/patches/ldmicroqt-2.3.2.patchset [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+diff --git a/ldmicro/CMakeLists.txt b/ldmicro/CMakeLists.txt
+index 58910f1..77f58fb 100644
+--- a/ldmicro/CMakeLists.txt
++++ b/ldmicro/CMakeLists.txt
+@@ -39,15 +39,6 @@ IF(UNIX)
+     execute_process(COMMAND lsb_release -cs
+     OUTPUT_VARIABLE RELEASE_CODENAME
+     OUTPUT_STRIP_TRAILING_WHITESPACE)
+-    STRING(TOLOWER ${RELEASE_CODENAME} RELEASE_LOWER)
```
Comment (korli) [inline sci-electronics/ldmicroqt/patches/ldmicroqt-2.3.2.patchset:15]: this block can go in a 
`if (not HAIKU)`

File: sci-electronics/ldmicroqt/ldmicroqt-2.3.2.recipe [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+CPUs from a ladder diagram."
+HOMEPAGE="https://github.com/FOSSEE/LDmicroQt"
+COPYRIGHT="2007 Jonathan Westhues"
+LICENSE="GNU GPL v1"
+REVISION="1"
+SOURCE_URI="https://github.com/FOSSEE/LDmicroQt/archive/refs/tags/$portVersion.tar.gz"
+CHECKSUM_SHA256="68e0469a5c907858289de1ee30ae14b2d13af695daa38f76c95aac1137d89139"
+SOURCE_DIR="LDmicroQt-$portVersion"
+PATCHES="ldmicroqt-$portVersion.patchset"
+ARCHITECTURES="all !x86_gcc2"
```
Comment (Begasus) [inline sci-electronics/ldmicroqt/ldmicroqt-2.3.2.recipe:13]: Empty line between PATCHES and ARCHITECTURES ...
```diff
-PATCHES="ldmicroqt-$portVersion.patchset"
-ARCHITECTURES="all !x86_gcc2"
+PATCHES="ldmicroqt-$portVersion.patchset"
+
+ARCHITECTURES="all !x86_gcc2"
```

**Comment by extrowerk** [comment]
You should add 	cmd:perl to the BUILDE_REQUIREMENTS section.

Build() should actually build:
```
	cd ldmicro/
	rm -rf build
	mkdir build
	cd build/
	cmake .. \
		-DCMAKE_BUILD_TYPE=Release
	make $jobArgs
```
It seems mkspecs insist to add -fPIE to the C/CXX flags which breaks the build on Haiku. I see no easy trivial way  to remove it, but sed'in the ```build/CMakeFiles/ldinterpret.dir/flags.make``` file can work.

After manually removing the problematic flags, the compilation breaks here:

```
[ 98%] Linking CXX executable ldmicro

differences follow:
    Only in expected: avr-hw.hex
    Only in expected: contacts.hex
    Only in expected: integers.hex
    Only in expected: misc-ops.hex
    Only in expected: pic16-hw.hex
    Only in expected: timers-counters.hex
(6 difference(s)/6)
FAIL
CMakeFiles/ldmicro.dir/build.make:608: recipe for target 'ldmicro' failed
make[2]: *** [ldmicro] Error 255
make[2]: *** Deleting file 'ldmicro'
CMakeFiles/Makefile2:127: recipe for target 'CMakeFiles/ldmicro.dir/all' failed
make[1]: *** [CMakeFiles/ldmicro.dir/all] Error 2
Makefile:155: recipe for target 'all' failed
make: *** [all] Error 2
/RAMDisk/LDmicroQt-2.3.2/ldmicro/b>
```

**Comment by Begasus** [comment]
Thanks to all involved :)

**Pull request merged**